document-portal/document-portal-fuse.c

```diff
@@ -165,14 +165,27 @@ G_DEFINE_AUTOPTR_CLEANUP_FUNC (XdpDomain, xdp_domain_unref)
 
 G_LOCK_DEFINE (domain_inodes);
 
+/* The reference count and use count are different. The latter still
+ * allow keeping track of the file without keeping it open, which otherwise
+ * keep the files around indefinitely.
+ * See https://github.com/flatpak/xdg-desktop-portal/issues/689
+ */
```

Comment:
Why not make a weak reference where you'd use the `use_count`? Would that solve anything? Ideally it wouldn't keep the file open, so it can still be freed up when nothing else references it.